Toolchain: Always strip, make code less confusing

BenWiederhake opened the pull request:
I always manually strip the executables, simply because a 1.2 GB toolchain is ridiculous and might cause some slowdowns during build.

So I thought: Hey, we have the code here anyway, why not enable it for everyone(*)? So that's what I did.

(*): Except MacOS users. Apparently there's something wonky, but I can't test it.

IdanHo commented:
Looks like you accidentally have #5740's commits included in this branch as well

BenWiederhake commented:
That's why I marked it as "draft" before I ran out of time for the day :)

BenWiederhake commented:
Changes since the first push:
- Fix build-breaking mistake, doink!
- Make build-error analysis less irritating

BenWiederhake commented:
Changes since the first push:
- Fix *other* build-breaking mistake, doink²!

ADKaster commented on .github/workflows/cmake.yml:
```diff
@@ -150,7 +150,7 @@ jobs:
       if: ${{ !cancelled() && matrix.debug-macros == 'NORMAL_DEBUG'}}
       working-directory: ${{ github.workspace }}/Build
-      run: cat ./debug.log
+      run: '[ ! -e debug.log ] || cat debug.log'
```
Hmm the point of this step is to make sure that we can see the boot dbg log even if the previous step which runs on target tests fails. If we don't generate the file at all because of "reasons" (maybe someone changes the run.sh step but forgot to update CI?) then this change would hide that, right? Maybe we could ferry some input/output from the `run tests` step? Not sure that's very elegant...

BenWiederhake replied:
You're right that this might "hide" an error if it fails at just the right moment. However, `run.sh` calls qemu with `-debugcon file:debug.log`, so this step has to fail extremely early for the file to be missing; in which case the emulation step will have generated interesting output and resulted in nonzero exit-code.

I'm not aware of a much better solution:
- If we fail here, then it "hides" the problem by being duplicious.
- If we add `touch debug.log` before the emulation, then this step will also "fail silently".

What do you suggest?

ADKaster replied:
Hmm. Yeah we'd have to be calling qemu incorrectly in order for it to hide the failure. Could make it not a separate step but just `ninja run || cat ./debug.log` with some `echo startgroup` magic in the middle? It's probably fine as is, that kind of error should be caught pretty quickly by a reviewer or by someone checking the logs.

BenWiederhake replied:
I like that idea!

BenWiederhake commented:
Changes since last push:
- Rebased
- Changed the "ninja run qemu magic" to fail if and only if qemu failed.

CC @ADKaster, now it uses the "group" magic, what do you think? :)

ADKaster reviewed (approved):
Looks like it'll do the trick! No annoying errors this way.
